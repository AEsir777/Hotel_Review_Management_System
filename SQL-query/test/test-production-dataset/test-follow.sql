/**** add friend ******/
/*** Test insert and isFriend ******/
-- check: not exist before
CALL GetFollowers('_3R2OM6fTkdcc3rPGTZYAQ'); -- the result shouldn't include '_4gdJdZFM0EnWr6v1Qr3A'
CALL GetFollowers('_4gdJdZFM0EnWr6v1Qr3A'); -- the result shouldn't include '_3R2OM6fTkdcc3rPGTZYAQ'
CALL GetFollowings('_3R2OM6fTkdcc3rPGTZYAQ'); -- the result shouldn't include '_4gdJdZFM0EnWr6v1Qr3A'
CALL GetFollowings('_4gdJdZFM0EnWr6v1Qr3A'); -- the result shouldn't include '_3R2OM6fTkdcc3rPGTZYAQ'
-- success case
CALL InsertFriend('_4gdJdZFM0EnWr6v1Qr3A', '_3R2OM6fTkdcc3rPGTZYAQ');
CALL InsertFriend('_3R2OM6fTkdcc3rPGTZYAQ', '_4gdJdZFM0EnWr6v1Qr3A');
-- check
CALL GetFollowers('_3R2OM6fTkdcc3rPGTZYAQ'); 
CALL GetFollowers('_4gdJdZFM0EnWr6v1Qr3A'); 
CALL GetFollowings('_3R2OM6fTkdcc3rPGTZYAQ');
CALL GetFollowings('_4gdJdZFM0EnWr6v1Qr3A'); 

CALL GetFollowers('_3R2OM6fTkdcc3rPGTZYAQ'); 
CALL GetFollowers('0rb0x3YydlvYdbFUysfsg');
CALL GetFollowings('_3R2OM6fTkdcc3rPGTZYAQ');
CALL GetFollowings('0rb0x3YydlvYdbFUysfsg');

-- success case 
CALL InsertFriend("0rb0x3YydlvYdbFUysfsg", "_3R2OM6fTkdcc3rPGTZYAQ");
CALL InsertFriend("_3R2OM6fTkdcc3rPGTZYAQ", "0rb0x3YydlvYdbFUysfsg");

-- check
CALL GetFollowers('_3R2OM6fTkdcc3rPGTZYAQ'); 
CALL GetFollowers('0rb0x3YydlvYdbFUysfsg');
CALL GetFollowings('_3R2OM6fTkdcc3rPGTZYAQ');
CALL GetFollowings('0rb0x3YydlvYdbFUysfsg');

-- check: exist before
CALL GetFollowers("_GvKig8yKIp_WtFtWpntkg");
CALL GetFollowings("qms0w0pmjjjaeNojF3qxQ");
CALL GetFollowers("_H6E5kheFYwu52kCZlF8qg"); 
CALL GetFollowings("1lmyWoPFb_eJvG5Zw3UA");
CALL GetFollowers("_iYyZtuWMmEMFCOKRLYgg"); 
CALL GetFollowings("pQrWjL22uvSWMwS9yZ_Yg");
CALL GetFollowers("_KeuaQCbJsZ7cipYpBPxDQ"); 
CALL GetFollowings("coWYR6JBcD9Qa55ibyqr9Q");
CALL GetFollowers("V3XYG1zKuW8VUBNtjacoxA");

-- check: insert fail
CALL InsertFriend("LwZJFLGxQwjjeOgpqTJnfw", "RHsmgIvCfnFL7DPNptKY8A");
CALL InsertFriend("c7XI431TnC9_AkmC-qrXvQ", "RgDVC3ZUBqpEe6Y1kPhIpw");
CALL InsertFriend("w-zD8Ln3XZszM82AfVrspg", "RgDVC3ZUBqpEe6Y1kPhIpw");
CALL InsertFriend("8Akq8sh1LSmJbqa6Jjxxqg", "96IKLbNHdK-sOAjuMwoegw");
CALL InsertFriend("LwZJFLGxQwjjeOgpqTJnfw", "abcd");
CALL InsertFriend("LwZJFLGxQwjjeOgpqTJnfw", "");