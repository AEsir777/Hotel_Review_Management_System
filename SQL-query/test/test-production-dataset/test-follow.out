/**** add friend ******/
/*** Test insert and isFriend ******/
-- check: not exist before
CALL GetFollowers('_3R2OM6fTkdcc3rPGTZYAQ');
output: 
NULL NULL
expected output:
NULL NULL
// the result shouldn't include '_4gdJdZFM0EnWr6v1Qr3A'

CALL GetFollowers('_4gdJdZFM0EnWr6v1Qr3A'); 
output: 
NULL NULL
expected output:
NULL NULL
// the result shouldn't include '_3R2OM6fTkdcc3rPGTZYAQ'

CALL GetFollowings('_3R2OM6fTkdcc3rPGTZYAQ'); 
output: 
NULL NULL
expected output:
NULL NULL
// the result shouldn't include '_4gdJdZFM0EnWr6v1Qr3A'

CALL GetFollowings('_4gdJdZFM0EnWr6v1Qr3A'); 
output: 
QngamgrqX9otZ6PUkd10Ig	Brittany
expected output:
QngamgrqX9otZ6PUkd10Ig	Brittany
// the result shouldn't include '_3R2OM6fTkdcc3rPGTZYAQ'

-- success case
CALL InsertFriend('_4gdJdZFM0EnWr6v1Qr3A', '_3R2OM6fTkdcc3rPGTZYAQ');
CALL InsertFriend('_3R2OM6fTkdcc3rPGTZYAQ', '_4gdJdZFM0EnWr6v1Qr3A');
2 row(s) affected
// expected output: 2 row(s) affected

-- check

CALL GetFollowers('_3R2OM6fTkdcc3rPGTZYAQ');
output: 
_4gdJdZFM0EnWr6v1Qr3A	Lauren
expected output:
_4gdJdZFM0EnWr6v1Qr3A	Lauren

CALL GetFollowers('_4gdJdZFM0EnWr6v1Qr3A');
output: 
_3R2OM6fTkdcc3rPGTZYAQ	Jessica
expected output:
_3R2OM6fTkdcc3rPGTZYAQ	Jessica

CALL GetFollowings('_3R2OM6fTkdcc3rPGTZYAQ');
output: 
_4gdJdZFM0EnWr6v1Qr3A	Lauren
expected output:
_4gdJdZFM0EnWr6v1Qr3A	Lauren

CALL GetFollowings('_4gdJdZFM0EnWr6v1Qr3A');
output: 
_3R2OM6fTkdcc3rPGTZYAQ	Jessica
QngamgrqX9otZ6PUkd10Ig	Brittany
expected output:
_3R2OM6fTkdcc3rPGTZYAQ	Jessica
QngamgrqX9otZ6PUkd10Ig	Brittany

CALL GetFollowers('_3R2OM6fTkdcc3rPGTZYAQ');
output: 
_4gdJdZFM0EnWr6v1Qr3A	Lauren
expected output:
_4gdJdZFM0EnWr6v1Qr3A	Lauren

CALL GetFollowers('0rb0x3YydlvYdbFUysfsg');
output: 
NULL NULL
expected output:
NULL NULL

CALL GetFollowings('_3R2OM6fTkdcc3rPGTZYAQ');
output: 
_4gdJdZFM0EnWr6v1Qr3A	Lauren
expected output:
_4gdJdZFM0EnWr6v1Qr3A	Lauren

CALL GetFollowings('0rb0x3YydlvYdbFUysfsg');
output: 
NULL NULL
expected output:
NULL NULL

-- success case
CALL InsertFriend("0rb0x3YydlvYdbFUysfsg", "_3R2OM6fTkdcc3rPGTZYAQ");
CALL InsertFriend("_3R2OM6fTkdcc3rPGTZYAQ", "0rb0x3YydlvYdbFUysfsg");
2 row(s) affected
// expected output: 2 row(s) affected

-- check
CALL GetFollowers('_3R2OM6fTkdcc3rPGTZYAQ'); 
output:
_4gdJdZFM0EnWr6v1Qr3A	Lauren
0rb0x3YydlvYdbFUysfsg	Lindsey
expected output:
_4gdJdZFM0EnWr6v1Qr3A	Lauren
0rb0x3YydlvYdbFUysfsg	Lindsey

CALL GetFollowers('0rb0x3YydlvYdbFUysfsg');
output:
_3R2OM6fTkdcc3rPGTZYAQ	Jessica
expected output:
_3R2OM6fTkdcc3rPGTZYAQ	Jessica

CALL GetFollowings('_3R2OM6fTkdcc3rPGTZYAQ');
output:
_4gdJdZFM0EnWr6v1Qr3A	Lauren
0rb0x3YydlvYdbFUysfsg	Lindsey
expected output:
_4gdJdZFM0EnWr6v1Qr3A	Lauren
0rb0x3YydlvYdbFUysfsg	Lindsey

CALL GetFollowings('0rb0x3YydlvYdbFUysfsg');
output:
_3R2OM6fTkdcc3rPGTZYAQ	Jessica
expected output:
_3R2OM6fTkdcc3rPGTZYAQ	Jessica


-- check: exist before
CALL GetFollowers("_GvKig8yKIp_WtFtWpntkg");
output:
7zbVroyptTaZUikxw6_dBw	Treva
expected output:
7zbVroyptTaZUikxw6_dBw	Treva

CALL GetFollowings("qms0w0pmjjjaeNojF3qxQ");
output:
_gXgA3Z3HXdXb9hq64iM3g	Diana
expected output:
_gXgA3Z3HXdXb9hq64iM3g	Diana

CALL GetFollowers("_H6E5kheFYwu52kCZlF8qg"); 
output:
t1BoPNoTINq2MhQo9pwYoQ	Matt
expected output:
t1BoPNoTINq2MhQo9pwYoQ	Matt

CALL GetFollowings("1lmyWoPFb_eJvG5Zw3UA");
output:
_hMI9tPsb2xmREGeLhDsIA	John
expected output:
_hMI9tPsb2xmREGeLhDsIA	John

CALL GetFollowers("_iYyZtuWMmEMFCOKRLYgg"); 
output:
pQrWjL22uvSWMwS9yZ_Yg	Sir Dennis
expected output:
pQrWjL22uvSWMwS9yZ_Yg	Sir Dennis

CALL GetFollowings("pQrWjL22uvSWMwS9yZ_Yg");
output:
_iYyZtuWMmEMFCOKRLYgg	Karen
expected output:
_iYyZtuWMmEMFCOKRLYgg	Karen

CALL GetFollowers("_KeuaQCbJsZ7cipYpBPxDQ"); 
output:
Xl5_VNVExSKmiv4uDY_DAA	Jonathan
expected output:
Xl5_VNVExSKmiv4uDY_DAA	Jonathan

CALL GetFollowings("coWYR6JBcD9Qa55ibyqr9Q");
output:
_JBQYhcgN16e3NK0l1Z5AQ	Leah
expected output:
_JBQYhcgN16e3NK0l1Z5AQ	Leah

CALL GetFollowers("V3XYG1zKuW8VUBNtjacoxA");
output:
2KyAPAsnGlmQbN1J8Eug	Amy
6QisM8Z_09QPEe__NIcNUw	Sara
7RZIOzLKugDvFjdJFwkHYw	Richard
AEsiF7xY6fL37pZgCAHLQ	Brian
aHxYkJqD7sHH6c0IflwLzA	Dave
IENZbosAEMTuD1Ob8mSIMg	Kristin
JJDTefsXi9n3nBj42heb_Q	Devon
Jqi2O_6tC8RpCg53zWeNw	Tom
KaL2ZkT6ppmuZnLR1yGAIw	Bill
oxigaAgicvvSwOLfd28zg	Danielle
expected output:
2KyAPAsnGlmQbN1J8Eug	Amy
6QisM8Z_09QPEe__NIcNUw	Sara
7RZIOzLKugDvFjdJFwkHYw	Richard
AEsiF7xY6fL37pZgCAHLQ	Brian
aHxYkJqD7sHH6c0IflwLzA	Dave
IENZbosAEMTuD1Ob8mSIMg	Kristin
JJDTefsXi9n3nBj42heb_Q	Devon
Jqi2O_6tC8RpCg53zWeNw	Tom
KaL2ZkT6ppmuZnLR1yGAIw	Bill
oxigaAgicvvSwOLfd28zg	Danielle

-- check: insert fail
CALL InsertFriend("LwZJFLGxQwjjeOgpqTJnfw", "RHsmgIvCfnFL7DPNptKY8A");
Error Code: 1062. Duplicate entry "LwZJFLGxQwjjeOgpqTJnfw"-"RHsmgIvCfnFL7DPNptKY8A" for key 'friend.PRIMARY'
// expected output: fail

CALL InsertFriend("c7XI431TnC9_AkmC-qrXvQ", "RgDVC3ZUBqpEe6Y1kPhIpw");
Error Code: 1452. Cannot add or update a child row: a foreign key constraint fails 
// expected output: fail

CALL InsertFriend("w-zD8Ln3XZszM82AfVrspg", "RgDVC3ZUBqpEe6Y1kPhIpw");
Error Code: 1452. Cannot add or update a child row: a foreign key constraint fails 
// expected output: fail

CALL InsertFriend("8Akq8sh1LSmJbqa6Jjxxqg", "96IKLbNHdK-sOAjuMwoegw");
Error Code: 1452. Cannot add or update a child row: a foreign key constraint fails 
// expected output: fail

CALL InsertFriend("LwZJFLGxQwjjeOgpqTJnfw", "abcd");
Error Code: 1452. Cannot add or update a child row: a foreign key constraint fails (`cs348`.`friend`, CONSTRAINT `friend_ibfk_2` FOREIGN KEY (`uid2`) REFERENCES `userfile` (`uid`))
// expected output: fail

CALL InsertFriend("LwZJFLGxQwjjeOgpqTJnfw", "");
Error Code: 1452. Cannot add or update a child row: a foreign key constraint fails (`cs348`.`friend`, CONSTRAINT `friend_ibfk_2` FOREIGN KEY (`uid2`) REFERENCES `userfile` (`uid`))
// expected output: fail